lib/IDE/CodeCompletion.cpp

```diff
@@ -4860,6 +4865,11 @@ void CodeCompletionCallbacksImpl::doneParsing() {
   Optional<Type> ExprType;
   ConcreteDeclRef ReferencedDecl = nullptr;
   if (ParsedExpr) {
+    if (auto *checkedExpr = findParsedExpr(CurDeclContext,
+                                           ParsedExpr->getSourceRange())) {
+      ParsedExpr = checkedExpr;
+    }
+
     if (auto typechecked = typeCheckParsedExpr()) {
```

Comment:
Should there be a prepareForRetypechecking here?

Reply:
`prepareForRetypechecking()` is done in `
typeCheckParsedExpr()`.
https://github.com/apple/swift/blob/82b5f2a19563341a87afbd2f574e7b80aecd08e1/lib/IDE/CodeCompletion.cpp#L1282-L1304